Question5.java

```diff
@@ -0,0 +1,191 @@
+/**
+ * Stack interface
+ */
+interface Stack<Type>
+{
+    void push(Type data) throws StackSizeExceeded;
+    void pop();
+    void peek();
+    int size();
+}
+/**
+ * Exception Classes
+ */
+class StackSizeExceeded extends Exception
+{
+    public StackSizeExceeded(String message) {super(message);}
+}
+class IllegalArgumentException extends Exception
+{
+    public IllegalArgumentException(String message) {super(message);}
+}
+
+/**
+ * Node class
+ */
+class Node<Type>
+{
+    Type data;
+    Node next;
+    /**
+     * Create node
+     * @param data
+     */
+    public Node(Type data)
+    {
+        this.data = data;
+        next = null;
+    }
+
+    public Node getNext() {
+        return next;
+    }
+
+    public Type getData() {
+        return data;
+    }
+
+    public void setNext(Node node) {
+        this.next = node;
+    }
+
+}
+/**
+ * LimitedStack Class
+ */
+class LimitedStack<Type> implements Stack<Type>
+{
+    private Node head;
+    private int maxStack;
+    private int stackSize;
+
+    public LimitedStack(int max) throws IllegalArgumentException
+    {
+        /**
+         * Validate Argument
+         */
+        if(max <= 0)
+        {
+            throw new IllegalArgumentException("Size cannot be {"+max+"}");
+        }
+        else {maxStack=max;stackSize=0;}
+    }
+
+    /**
+     * Push method
+     * @param data
+     */
+    @Override
+    public void push(Type data) throws StackSizeExceeded
+    {
+        if (maxStack>0)
+        {
+            Node<Type> node = new Node<>(data);
+
+            if(head==null) {
+                head = node;
+            }
+
+            else
+            {
+                node.setNext(head);
+                head = node;
+            }
+            maxStack--;
```

Comment:
Dont do this. Keep the size constant. Add check basis the current size

Reply:
This issue has been resolved.